### DISCUSSION TITLE
Multiobject custom field in site not getting ip address value

### DISCUSSION THREAD

**Discussion by neharaina1771** [netbox-community/netbox#15863] [Help Wanted!] [answered]
Hi

I am using pynetbox to save the IP Address in IPAM .I am trying to assign the IP address to custom field in site which is multiobject IP address object type.Although IPs are getting displayed under custom field but when I try to assign the IPs using pynetbox to custom field It is failing..'

site = pynetbox.dcim.sites.get(name="site-name")

netbox_ip = pynetbox.api.ipam.ip_addresses.create(address="ip_address", status="active", tenant=tenanat.id)
netbox_ip.assigned_object = site
netbox_ip.assigned_object_id = site.id
netbox_ip.assigned_object_type = 'dcim.site'
netbox_ip.save()

Error:- 400 Bad request {'assigned_object_type': ["Invalid value".Specify a content type as '<app_label:<model_name>>']}

**Comment by neharaina1771** [comment]
Also my goal is to assign IP address to custom fields for multiple sites via api which is multiobject IP adresss type.How can I achieve that efficiently as there are 600 sites.

**Comment by candlerb** [comment]
> Error:- 400 Bad request {'assigned_object_type': ["Invalid value".Specify a content type as '<app_label:<model_name>>']}

That seems to be telling you to try `dcim:site` instead of `dcim.site`

But I didn't think you could assign an IP address directly to a site. I thought you had to assign it to an interface or vminterface.

Also, the code you showed doesn't seem to do anything with custom fields. If you have created a custom field on IPAddress, then it would be something like

```
pynetbox.api.ipam.ip_addresses.create(address="ip_address", status="active", tenant=tenanat.id, custom_fields={"foo":"bar"})
```

However, since you said you created a custom field on Site, then clearly you'd have to update the custom fields on the specific site of interest.

I suggest you try making some object assignments within the GUI, and use pynetbox to query the site before and after, to see how the custom field data has changed.

Here, for example, I  have some custom fields set on Device:

```
>>> a = nb.dcim.devices.get(name="nuc1")
>>> type(a)
<class 'pynetbox.models.dcim.Devices'>
>>> a.custom_fields
{'monitoring': ['node'], 'snmp_auth': None, 'snmp_module': None}
>>>
```

You should be able to read/update custom fields on Site in a similar way.

**Reply by neharaina1771** [reply]
Thanks for the reply.
Sorry my bad Error:- 400 Bad request {'assigned_object_type': ["Invalid value".Specify a content type as '<app_label:<model_name>>']} it is '<app_label.<model_name> dot instead of colon

I have created the custom field on Site which is multiobject field for IP address.From UI I am able to select the IP address for the custom field but using pynetbox I am not able to 
site - pynetbox.dcim.sites.get(name="mysite")
site.custom_fields["key"]= value
site.saves()

then it gives me error saying that 400 Bad request {'custom_fields': [Unknown related objects]: 10.4.128.160/27}

**Reply by neharaina1771** [reply]
I am able to implement it ..but the values is not showing up for custom field which is multiobject ..Rest fields like text etc are saved successfully .just seems to be case with multiobject IP address fields
>>> a = nb.dcim.devices.get(name="nuc1")
>>> type(a)
<class 'pynetbox.models.dcim.Devices'>
>>> a.custom_fields
{'monitoring': ['node'], 'snmp_auth': None, 'snmp_module': None}

**Reply by candlerb** [reply, marked as answer]
> 400 Bad request {'custom_fields': [Unknown related objects]: 10.4.128.160/27}

I suspect you need to give a list of object IDs: i.e. first do a query to lookup IP address 10.4.128.160/27 and get its ID.

**Reply by neharaina1771** [reply]
I assigned IP address from UI and then get the site custom field value it is giving

{'test': {'id':"34088", 'url': 'https://localhost:8000/api/ipam/ip-addresses/34088/', 'display': '10.4.128.160/27', 'family': 4, 'address': '10.4.128.160/27''}}

test is custom field on site and type of site is <class 'pynetbox.core.response.Record'>

**Reply by neharaina1771** [reply]
{'test': [{'id':"34088", 'url': 'https://localhost:8000/api/ipam/ip-addresses/34088/', 'display': '10.4.128.160/27', 'family': 4, 'address': '10.4.128.160/27''}]

output is inside list

**Reply by neharaina1771** [reply]
Thank you for the help ..

**Reply by candlerb** [reply]
When updating this, I would expect you could send `custom_fields={"test": [34088]}` or `custom_fields={"test": [{"id":34088}]}`

It might even be possible to do `custom_fields={"test":[{"address":"10.4.128.160/27"}]}`

You'll have to see which one works.
